File: android/src/main/java/com/ammarahmed/mmkv/IDStore.java
package com.ammarahmed.mmkv;

import android.os.Bundle;

import androidx.annotation.Nullable;
import com.tencent.mmkv.MMKV;
import java.util.HashMap;

public class IDStore {

    private MMKV store;


    public IDStore(MMKV kv) {

        store = kv;
    }


    public void add(String ID, boolean encrypted, @Nullable String alias) {

        boolean hasKey = store.containsKey("mmkvIdStore");
        HashMap<String, Object> IdStore = new HashMap<>();
        if (hasKey) {
            Bundle mmkvIdStore = store.decodeParcelable("mmkvIdStore", Bundle.class);
            IdStore = (HashMap<String, Object>) mmkvIdStore.getSerializable("mmkvIdStore");

        }

        HashMap<String, Object> child = new HashMap<>();
        child.put("ID", ID);
        child.put("encrypted", encrypted);
        child.put("alias", alias);
        IdStore.put(ID, child);

        Bundle bundle = new Bundle();
        bundle.putSerializable("mmkvIdStore", IdStore);

        store.encode("mmkvIdStore", bundle);

    }

    public boolean encrypted(String ID) {

        boolean hasKey = store.containsKey("mmkvIdStore");
        HashMap<String, Object> IdStore = new HashMap<>();

        if (!hasKey) {
            return false;
        }
        Bundle mmkvIdStore = store.decodeParcelable("mmkvIdStore", Bundle.class);
        IdStore = (HashMap<String, Object>) mmkvIdStore.getSerializable("mmkvIdStore");

        HashMap<String, Object> child = (HashMap<String, Object>) IdStore.get(ID);


        return (boolean) child.get("encrypted");


    }

    public String getAlias(String ID) {

        boolean hasKey = store.containsKey("mmkvIdStore");
        HashMap<String, Object> IdStore = new HashMap<>();

        if (!hasKey) {
            return null;
        }
        Bundle mmkvIdStore = store.decodeParcelable("mmkvIdStore", Bundle.class);
        IdStore = (HashMap<String, Object>) mmkvIdStore.getSerializable("mmkvIdStore");

        HashMap<String, Object> child = (HashMap<String, Object>) IdStore.get(ID);

        return (String) child.get("alias");

    }


    public HashMap<String, Object> getAll() {

        boolean hasKey = store.containsKey("mmkvIdStore");
        HashMap<String, Object> IdStore = new HashMap<>();

        if (!hasKey) {
            return null;
        }
        Bundle mmkvIdStore = store.decodeParcelable("mmkvIdStore", Bundle.class);
        IdStore = (HashMap<String, Object>) mmkvIdStore.getSerializable("mmkvIdStore");


        return IdStore;

    }


    public boolean exists(String ID) {
        boolean hasKey = store.containsKey("mmkvIdStore");
        if (hasKey) {
            HashMap<String, Object> IdStore = new HashMap<>();
            Bundle mmkvIdStore = store.decodeParcelable("mmkvIdStore", Bundle.class);
            IdStore = (HashMap<String, Object>) mmkvIdStore.getSerializable("mmkvIdStore");


            return IdStore.containsKey(ID);

        } else {

            return false;
        }

    }

}

File: ios/IDStore.m
//
//  NSObject+IDStore.m
//  DoubleConversion
//
//  Created by Ammar Ahmed on 1/10/1399 AP.
//


#import "IDStore.h"
#import <MMKV/MMKV.h>

@implementation IDStore: NSObject

MMKV *store;

NSString *storeId = @"mmkvIdStore";
NSString *storeData = @"mmkvIdData";

- (id) initWithMMKV:(MMKV *)kv{
    
    self = [super init];
    store = kv;
    
    return self;
}



- (void)add:(NSString *)ID
  encrypted:(bool)encrypted
      alias:(NSString *)alias
{
    
    
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    
    if ([store containsKey:storeData]) {
        dic = [store getObjectOfClass:NSMutableDictionary.class forKey:storeData];
    }
    
    NSMutableDictionary *properties = [NSMutableDictionary dictionary];
    
    [properties setValue:[NSNumber numberWithBool:encrypted] forKey:@"encrypted"];
    
    
    [properties setValue:alias forKey:@"alias"];
    
    
    [properties setValue:ID forKey:@"id"];
    [dic setObject:properties forKey:ID];
    
    [store setObject:dic forKey:storeData];
    
}

- (bool) encrypted:(NSString *)ID {
    
    NSMutableDictionary *dic = [store getObjectOfClass:NSMutableDictionary.class forKey:storeData];
    
    NSMutableDictionary *properties = [dic objectForKey:ID];
    
    return [[properties valueForKey:@"encrypted"] boolValue];
    
    
}

- (NSString *) getAlias:(NSString *)ID {
    
    NSMutableDictionary *dic = [store getObjectOfClass:NSMutableDictionary.class forKey:storeData];
    
    NSMutableDictionary *properties = [dic objectForKey:ID];
    
    return [properties valueForKey:@"alias"];
}

- (NSMutableDictionary *) getProperties:(NSString *)ID {
    
    NSMutableDictionary *dic = [store getObjectOfClass:NSMutableDictionary.class forKey:storeData];
    
    return [dic objectForKey:ID];
}



- (NSMutableDictionary *)getAll {
    if ([store containsKey:storeData]) {
        
        return [store getObjectOfClass:NSMutableDictionary.class forKey:storeData];
        
    } else {
        
        return NULL;
    }
}



-(bool)exists:(NSString *)ID {
    
    bool hasKey = [store containsKey:storeData];
    
    if (hasKey) {
           NSMutableDictionary *mmkvStore = [store getObjectOfClass:NSMutableDictionary.class forKey:storeData];
        
        return [[mmkvStore allKeys] containsObject:ID];
        
    } else {
        return false;
    }
    
}


@end
